### ISSUE TITLE
CMUS not recognizing my TRCK tags (CMUS plays track in alphanumeric order even though TRCK and TPOS is set for mp3 files)

### ISSUE THREAD

**Issue by jonascj** [cmus/cmus#594]
tl;dr;
-----
According to `mid3v2 -l`the mp3 files in question have tags like `TPOS=1/2` and `TRCK=1` but they are still shown as `0. Some Track Title` in view `1:tree`and `2:sort`. It gives me weird playback ordering. Whats going on?

Longer version
------------------

I have several MP3 albums which are played in alphabetical order according to track title in the TIT2 tag, even though TRCK and TPOS tags are set. I can't figure out why this is.

Below (*Track / tag examples*) I've included `mid3v2 -l` output for five tracks to exemplify the tag information and the play order, comment on the play order in next paragraph. Call the mp3 files 1-5 like the list below.

When starting CMUS I get dropped in view `1:tree`, then select the `Thin Lizzy` > `Greatest Hits` and it starts playing the files in order `1 > 2 > 3 > other tracks > 4` (see the numbered files below). This corresponds to playing them alphabetically by their TIT2 tag, not by TRCK or even filename. That is also the order of the tracks you find in view `2:sorted`, except that list starts with `0. Angel of Death` (because the title starts with `A` I assume), and `0. The ...` tracks come later in the list.

In view `1:tree` the tracks are listed according to their track number, or maybe by their filename since they list the same, but all tracks are shown as `0. Some Track Title`, e.g. the first displayed item in `1:tree` for this album is `0. The Boys Are Back In Town`.

Several questions:
 - I do not understand why CMUS displays all track numbesr as `0.`. I think this is likely the root of the problem. Are my ID3v2 tags wrong or what?
 - Why are the tracks sorted in the right order in `1:tree`? 
 - Why are they sorted wrongly in `2:sorted`? They should becmus sorted according to the option/var `lib_sort` which is set to `albumartist date album discnumber tracknumber title filename play_count` on my system (the default I guess).

I almost refuse to believe it is a bug, I must be me doing something n00bish. I've just started with CMUS coming from Guayadeque.

Track / tag examples
--------------------
1) `mid3v2 -l "101  The Boys Are Back In Town.mp3"`
```
IDv2 tag info for 101  The Boys Are Back In Town.mp3
COMM==XXX=C4
TALB=Greatest Hits
TCON=Rock
TDRC=2004
TIT2=The Boys Are Back In Town
TPE1=Thin Lizzy
TPOS=1/2
TRCK=1
```

2) `mid3v2 -l "217  The Boys Are Back In Town.mp3"`
```
IDv2 tag info for 217  The Boys Are Back In Town.mp3
TALB=Greatest Hits
TCON=Rock
TDRC=2004
TIT2=The Boys Are Back In Town
TPE1=Thin Lizzy4
TPOS=2/2
TRCK=17
```
3) `mid3v2 -l "211  The Rocker.mp3"`
```
IDv2 tag info for 211  The Rocker.mp3
TALB=Greatest Hits
TCON=Rock
TDRC=2004
TIT2=The Rocker
TPE1=Thin Lizzy
TPOS=2/2
TRCK=11
USLT=[unrepresentable data]
```
4) `mid3v2 -l "119  Yellow Pearl.mp3"`
```
IDv2 tag info for 119  Yellow Pearl.mp3
TALB=Greatest Hits
TCON=Rock
TDRC=2004
TIT2=Yellow Pearl
TPE1=Thin Lizzy
TPOS=1/2
TRCK=19
```
5) `mid3v2 -l "102  Jailbreak.mp3"`
```
IDv2 tag info for 102  Jailbreak.mp3
TALB=Greatest Hits
TCON=Rock
TDRC=2004
TIT2=Jailbreak
TPE1=Thin Lizzy
TPOS=1/2
TRCK=2
```

System info
-------------- 

1) `uname -a`:
```
Linux laptop 4.8.13-1-ARCH #1 SMP PREEMPT Fri Dec 9 07:24:34 CET 2016 x86_64 GNU/Linux
```

2) `cmus --version`:
```
cmus v2.7.1
```

3) `cmus --plugins`:
```
Input Plugins: /usr/lib/cmus/ip
  mad:
    Priority: 55
    File Types: mp3 mp2
    MIME Types: audio/mpeg audio/x-mp3 audio/x-mpeg
  flac:
    Priority: 50
    File Types: flac fla
    MIME Types:
  modplug:
    Priority: 50
    File Types: mod s3m xm it 669 amf ams dbm dmf dsm far mdl med mtm okt ptm stm ult umx mt2 psm
    MIME Types:
  wav:
    Priority: 50
    File Types: wav
    MIME Types:
  mp4:
    Priority: 50
    File Types: mp4 m4a m4b
    MIME Types:
  vorbis:
    Priority: 50
    File Types: ogg oga ogx
    MIME Types: application/ogg audio/x-ogg
  wavpack:
    Priority: 50
    File Types: wv
    MIME Types: audio/x-wavpack
  cue:
    Priority: 50
    File Types:
    MIME Types: application/x-cue
  cdio:
    Priority: 50
    File Types:
    MIME Types: x-content/audio-cdda
  opus:
    Priority: 50
    File Types: opus
    MIME Types:
  aac:
    Priority: 50
    File Types: aac
    MIME Types: audio/aac audio/aacp

Output Plugins: /usr/lib/cmus/op
  pulse
  alsa
  oss
  ao

```
edit: plugin information was scrambled by unintended middle-mouse pastes.

**Comment by mahkoh** [comment]
I think you have play_sorted enabled. Press `o` and watch the bottom right corner. Disabling this should play the tracks in the order they appear in the tree view.

**Comment by jonascj** [comment]
That seemed to be the issue. Thanks, and sorry for opening a non-issue. Closing unless the problem reappears.
